mage_ai/orchestration/db/models/schedules.py

```diff
@@ -824,13 +824,11 @@ def pipeline(self) -> 'Pipeline':
         if project_platform_activated():
             return self.pipeline_project_platform
 
-        return Pipeline.get(self.pipeline_uuid)
+        return Pipeline.get(self.pipeline_uuid, repo_path=get_repo_path())
 
     @property
     def pipeline_type(self) -> PipelineType:
-        pipeline = Pipeline.get(self.pipeline_uuid, check_if_exists=True)
-
-        return pipeline.type if pipeline is not None else None
+        return self.pipeline.type if self.pipeline is not None else None
```

Comment:
the code will fetch the pipeline twice. You can keep the `pipeline = self.pipeline`.